base/headers.go

```diff
@@ -12,6 +12,27 @@ import "strings"
 // Whitespace recognised by SIP protocol.
 const c_ABNF_WS = " \t"
 
+// Maybestring contains a string, or nil.
+type MaybeString interface {
+	implementsMaybeString()
+}
+
+// NoString represents the absence of a string.
+type NoString struct{}
+
+func (n NoString) implementsMaybeString() {}
+
+// String represents an actual string.
+type String struct {
+	S string
```

Comment:
If this is done, then other packages can't create a string using "base.String{"something"} syntax.

There would need to be a function to create them.  Something like "base.NewString("something")".  How do we feel about that?